drivers/ipm/ipm_mcux.c

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2017, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <device.h>
+#include <ipm.h>
+#include <fsl_lpc_mailbox.h>
+#include <fsl_clock.h>
+#include <soc.h>
+
+#define MCUX_IPM_DATA_REGS 1
+#define MCUX_IPM_MAX_ID_VAL 0
+
+
+struct mcux_mailbox_config {
+	MAILBOX_Type *base;
+	void (*irq_config_func)(struct device *dev);
+};
+
+struct mcux_mailbox_data {
+	ipm_callback_t callback;
+	void *callback_ctx;
+};
+
+static void mcux_mailbox_isr(void *arg)
+{
+	struct device *dev = arg;
+	struct mcux_mailbox_data *data = dev->driver_data;
+	mailbox_cpu_id_t cpu_id;
+#if defined(__CM4_CMSIS_VERSION)
```

Comment:
Rather than having multiple ifdef blocks, can we just do something at the top of this file like:

#if defined(__CM4_CMSIS_VERION)
  #define MAILBOX_ID kMAILBOX_CM4
#else
  #define MAILBOX_ID kMAILBOX_CM0
#endif

Reply:
ok